django/core/cache/backends/memcached.py

```diff
@@ -105,14 +105,14 @@ def close(self, **kwargs):
         self._cache.disconnect_all()
 
     def incr(self, key, delta=1, version=None):
-        # memcached doesn't support a negative delta
-        if delta < 0:
-            return self.decr(key, -delta, version=version)
         key = self.make_key(key, version=version)
         self.validate_key(key)
         try:
-            val = self._cache.incr(key, delta)
-
+            # memcached doesn't support a negative delta
+            if delta < 0:
+                val = self._cache.decr(key, -delta)
+            else:
+                val = self._cache.incr(key, delta)
         # Normalize an exception raised by the underlying client library to
         # ValueError in the event of a nonexistent key when calling incr().
```

Comment:
This comment should be amended to reflect that it is a call to `incr()` or `decr()` of the underlying client now they're in the same block.

Reply:
Updated.